### ISSUE TITLE
Unable to install firebase SDK 

### ISSUE THREAD

**Issue by hamzahsn** [firebase/firebase-js-sdk#776]
Recently with: 
OS: Windows 7 X64
node: v10.0.0
npm: 6.0.0
angular: 5.2.0

Trying to install firebase sdk the latest with angular 5 project with: npm 

> install firebase --save

 but I think it's not going well with gyp module in this version of node and it throw this error:  

```

> grpc@1.10.1 install C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc
> node-pre-gyp install --fallback-to-build --library=static_library

node-pre-gyp ERR! Tried to download(403): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.10.1/node-v64-win32-x64-unknown.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for grpc@1.10.1 and node@10.0.0 (node-v64 ABI, unknown) (falling back to source compile with node-gyp)
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\ROOT\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\ROOT\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:508:16)
gyp ERR! stack     at C:\Users\ROOT\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:149:21)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\ROOT\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-
build" "--library=static_library" "--module=C:\\Users\\ROOT\\Desktop\\TestingNewVersion\\node_modules\\@firebase\\firest
ore\\node_modules\\grpc\\src\\node\\extension_binary\\node-v64-win32-x64-unknown\\grpc_node.node" "--module_name=grpc_node" "--module_path=C:\\Users\\ROOT\\Desktop\\TestingNewVersion\\node_modules\\@firebase\\firestore\\node_modules\\grpc\\src\\node\\extension_binary\\node-v64-win32-x64-unknown"
gyp ERR! cwd C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\ROOT\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --
fallback-to-build --library=static_library --module=C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firesto
re\node_modules\grpc\src\node\extension_binary\node-v64-win32-x64-unknown\grpc_node.node --module_name=grpc_node --module_path=C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc\src\node\extension_binary\node-v64-win32-x64-unknown' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\fi
restore\node_modules\grpc\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:947:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:236:5)
node-pre-gyp ERR! System Windows_NT 6.1.7601
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\ROOT\\Desktop\\TestingNewVersion\\node_modul
es\\@firebase\\firestore\\node_modules\\grpc\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
node-pre-gyp ERR! cwd C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc
node-pre-gyp ERR! node -v v10.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.7.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\ROOT\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --library=sta
tic_library --module=C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc\src\node\e
xtension_binary\node-v64-win32-x64-unknown\grpc_node.node --module_name=grpc_node --module_path=C:\Users\ROOT\Desktop\TestingNewVersion\node_modules\@firebase\firestore\node_modules\grpc\src\node\extension_binary\node-v64-win32-x64-unknown' (1)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.10.1 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the grpc@1.10.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ROOT\AppData\Roaming\npm-cache\_logs\2018-05-04T07_24_23_994Z-debug.log

```

but when trying to install another version of firebase such as: firebase@4.6.0 it worked fine.

**Comment by hamzahsn** [comment]
Thanks, but I've already installed python and set the environment variable, but I think that the problem is with this version of node, because it worked fine with other versions of nodejs!
beside, I think that the core member are catching the gyp error now and trying to fix that

**Comment by phibonacci85** [comment]
has anyone found a solution to this yet? I'm getting the same problem... It almost seems like it's having a problem connecting to the googleapi site. maybe a bad ssl certificate or something??

**Comment by Swordys** [comment]
Had the same problem on my machine.
OS: Windows 10
Node: 10.0
npm: 6.0.1

Fixed it by switching to an LTS version of node (8.11.1) using nvm.

**Comment by hamzahsn** [comment]
@phibonacci85  As already mention, I think that the core member of gyp node are catching all the gyp errors, but If you install the LTS version as I did and as @Swordys did then I hope it will work for you, If not then try to pass your problem here and we will try to find a solution for that!

**Comment by buffos** [comment]
Same problem here (with windows 10 and node 10 , both npm and yarn)
  * will not get pre-compiled version from googleapis
  * exit from MSBuild.exe error code 1 (tried both 2017 and 2015 with no luck)

**Comment by hamzahsn** [comment]
@buffos As I mentioned, if you try to install another version of firebase sdk older than the recent ones with node 10 and Win10, then it worked fine, but the problem with the latest versions of firebase sdk with node 10. 
So I suggest to:
Try to install node8.11.1 LTS then try to install the latest version of firebase and if there is a problem you can pass it here.

**Comment by buffos** [comment]
@hamzahsn Installing node8 does not solve the current problem. In the end we might want to use node10

I am giving feedback to **what works** and **what does not** work in _node 10_, in order to help to fix for node 10

The fact that `firebase@4.12.1` works in node10, can surely help find and fix the problem.

**Comment by mmermerkaya** [comment]
This should be fixed by #785. The new version (that includes this PR) will most likely be released on Thursday.
